airflow/providers/hashicorp/hooks/vault.py

```diff
@@ -148,9 +149,29 @@ def __init__(
             except ValueError:
                 raise VaultError(f"The version is not an int: {conn_version}. ")
 
-        if auth_type in ["approle", "aws_iam"]:
+        if auth_type == "approle":
+            if not role_id:
+                if self.connection.extra_dejson.get('role_id'):
+                    role_id = self.connection.extra_dejson.get('role_id')
+                else:
+                    role_id = self.connection.login
+            else:
+                warnings.warn(
+                    """The usage of role_id has been deprecated.
+                    Please use either the connection login or extras.""",
```

Comment:
```suggestion
                    Please use connection login.""",
```

See my comment below

Reply:
Ok, so, to clarify.  No deprecation of `role_id` for `aws_am` and a modified warning for `AppRole`?